--- pip-18.0.orig/src/pip/_internal/wheel.py	2018-07-17 05:26:00.000000000 -0300
+++ pip-18.0/src/pip/_internal/wheel.py	2018-07-23 17:12:51.387197601 -0300
@@ -282,6 +282,18 @@
                 # to ensure we don't install empty dirs; empty dirs can't be
                 # uninstalled.
                 ensure_dir(destdir)
+                # Ensure that the destination file does not exist. This is
+                # required so that we break a hard link instead of replacing
+                # the contents in place
+                # xref: https://github.com/conda/conda/issues/6861
+                if os.path.lexists(destfile):
+                    os.unlink(destfile)
+                # Ensure that the destination file does not exist. This is
+                # required so that we break a hard link instead of replacing
+                # the contents in place
+                # xref: https://github.com/conda/conda/issues/6861
+                if os.path.lexists(destfile):
+                    os.unlink(destfile)
 
                 # copyfile (called below) truncates the destination if it
                 # exists and then writes the new contents. This is fine in most
